snap/snapcraft.yaml

```diff
@@ -43,9 +53,921 @@ apps:
+      - WORK_DIR: "/tmp/build-${PKGNAME}"
+      - PKGFMT: "deb"
+      - ARCH: "amd64"
+      - BUILD_PATH: ${OUTPUT_DIR}/${PKGFILE}
+      - PACKAGE: ${PKGNAME}_${PKGVERSION}-${ITERATION}_${ARCH}.${PKGFMT}
+    override-build: |
+      set -x
+      snapcraftctl build
+
+      gem install fpm
+      fpm \
+      -s dir \
+      -t "${PKGFMT}" \
+      -a "${ARCH}" \
+      -n "${PKGNAME}" \
+      -v "${PKGVERSION}" \
+      --iteration "${ITERATION}" \
+      --provides "${PKGNAME}" \
+      --conflicts "${PKGNAME}" \
+      --replaces "${PKGNAME}" \
+      --package  ${PACKAGE} \
+      -C $SNAPCRAFT_PART_INSTALL \
+      --description "ASN.1 (Release 15) compiler with OpenAirInterface (OAI)
+      specific changes. ASN.1 to C compiler takes the ASN.1 module files (example)
+      and generates the C++ compatible C source code. That code can be used to
+      serialize the native C structures into compact and unambiguous
+      BER/XER/PER-based data files, and deserialize the files back."
+
+      mv ${PACKAGE} $SNAPCRAFT_PART_INSTALL/
+
+  oai-gnutls:
+    plugin: autotools
+    after:
+      - oai-nettle
+    source: http://mirrors.dotsrc.org/gcrypt/gnutls/v3.1/gnutls-3.1.23.tar.xz
+    source-type: tar
+    autotools-configure-parameters:
+      - --with-libnettle-prefix=/root/parts/oai-nettle/install/usr/local/
+    build-packages:
+      - libtasn1-6-dev
+      - libp11-kit-dev
+      - libtspi-dev
+      - libtspi1
+      - libidn2-0-dev
+      - libidn11-dev
+      - ruby
+      - ruby-dev
+      - rubygems
+    build-environment:
+      - PKGNAME: "oai-gnutls"
+      - ITERATION: "1"
+      - PKGVERSION: "3.1.23"
+      - WORK_DIR: "/tmp/build-${PKGNAME}"
+      - PKGFMT: "deb"
+      - ARCH: "amd64"
+      - BUILD_PATH: ${OUTPUT_DIR}/${PKGFILE}
+      - PACKAGE: ${PKGNAME}_${PKGVERSION}-${ITERATION}_${ARCH}.${PKGFMT}
+    override-build: |
+      set -x
+      patch gl/stdio-impl.h <<'EOF'
+      @@ -18,6 +18,9 @@
+          the same implementation of stdio extension API, except that some fields
+          have different naming conventions, or their access requires some casts.  */
+
+      +#if !defined _IO_IN_BACKUP && defined _IO_EOF_SEEN
+      +# define _IO_IN_BACKUP 0x100
+      +#endif
+
+       /* BSD stdio derived implementations.  */
+
+      EOF
+      patch gl/fseterr.c <<'EOF'
+      @@ -29,7 +29,7 @@
+         /* Most systems provide FILE as a struct and the necessary bitmask in
+            <stdio.h>, because they need it for implementing getc() and putc() as
+            fast macros.  */
+      -#if defined _IO_ftrylockfile || __GNU_LIBRARY__ == 1 /* GNU libc, BeOS, Haiku, Linux libc5 */
+      +#if defined _IO_EOF_SEEN || __GNU_LIBRARY__ == 1 /* GNU libc, BeOS, Haiku, Linux libc5 */
+         fp->_flags |= _IO_ERR_SEEN;
+       #elif defined __sferror || defined __DragonFly__ /* FreeBSD, NetBSD, OpenBSD, DragonFly, Mac OS X, Cygwin */
+         fp_->_flags |= __SERR;
+      EOF
```

Comment:
Can we move this patch to its own file? This will be so easy to break inside the yaml.

Reply:
Fixed